packages/riverpod_annotation/lib/src/riverpod_annotation.dart

```diff
@@ -23,6 +24,11 @@ class Riverpod {
   /// Defaults to false.
   final bool keepAlive;
 
+  /// To set a custom name for the provider you want to generate.
+  ///
+  /// Defaults to null.
+  final String? name;
```

Comment:
```suggestion
  /// A custom name for the provider you want to generate.
  /// Riverpod adds a suffix to this, which defaults to `Provider`. Set the `provider_name_suffix` option in build_runner settings to change this; see [riverpod_generator](https://pub.dev/packages/riverpod_generator#global-configuration) for more details.
  ///
  /// Defaults to the name of the annotated method.
  final String? name;
```

The documentation for this is a bit unclear.